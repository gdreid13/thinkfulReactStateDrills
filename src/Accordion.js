import React from 'react';

class Accordion extends React.Component {
    static defaultProps = {
        sections: [{
            title: 'Default title',
            content: 'Default content',
        }]
    };
    state = {
        currentAccordIndex: 0
    };

    handleButtonClick(index) {
        this.setState ({
            currentAccordIndex: index
        })
    }
    
    renderButtons() {
        return this.props.sections.map((sections, index) => (
            <li key={index}>
                <button key={index} onClick={() => this.handleButtonClick(index)}>
                    {sections.title}
                </button>
                {this.renderContent(index)}
            </li>
        ))
    }

    renderContent(index) {
        const currentAccord = this.props.sections[this.state.currentAccordIndex]
        if (index === this.state.currentAccordIndex) {
            return (
                <p className='content'>
                {currentAccord.content}
                </p>
            )
        }
    }

    render () {
        return (
            <div>
                <ul>
                    {this.renderButtons()}
                </ul>
            </div>
        )
    }
}

export default Accordion;