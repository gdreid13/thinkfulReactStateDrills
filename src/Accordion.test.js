import React from 'react';
import ReactDOM from 'react-dom';
import Accordion from './Accordion';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Accordion component', () => {
    const accordProp = [
        {
          title: 'Section 1',
          content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
          title: 'Section 2',
          content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
          title: 'Section 3',
          content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
    ];

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Accordion />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders empty with no sections', () => {
        const wrapper = shallow(<Accordion />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders no sections active by default', () => {
        const wrapper = shallow(<Accordion sections={accordProp} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('opens clicked tab', () => {
        const wrapper = shallow(<Accordion sections={accordProp} />)
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('opens only last section when multiple sections clicked', () => {
        const wrapper = shallow(<Accordion sections={accordProp} />)
        wrapper.find('button').at(0).simulate('click')
        wrapper.find('button').at(1).simulate('click')
        wrapper.find('button').at(2).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot
    })
})