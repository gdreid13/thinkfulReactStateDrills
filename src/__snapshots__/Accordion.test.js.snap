// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component opens clicked tab 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      key="1"
    >
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
      <p
        className="content"
      >
        Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
      </p>
    </li>
    <li
      key="2"
    >
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;

exports[`Accordion component renders empty with no sections 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        key="0"
        onClick={[Function]}
      >
        Default title
      </button>
      <p
        className="content"
      >
        Default content
      </p>
    </li>
  </ul>
</div>
`;

exports[`Accordion component renders no sections active by default 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
      <p
        className="content"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
      </p>
    </li>
    <li
      key="1"
    >
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li
      key="2"
    >
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;
